pyuploadcare/resources/file_group.py

```diff
@@ -0,0 +1,209 @@
+import re
+
+import dateutil.parser
+
+from pyuploadcare import conf
+from pyuploadcare.exceptions import InvalidParamError
+from pyuploadcare.resources.base import ApiMixin
+from pyuploadcare.resources.file import File
+
+
+GROUP_ID_REGEX = re.compile(
+    r"""
+    (?P<group_id>
+        [a-z0-9]{8}-(?:[a-z0-9]{4}-){3}[a-z0-9]{12}
+        ~
+        (?P<files_qty>\d+)
+    )
+""",
+    re.VERBOSE,
+)
+
+
+class FileGroup(ApiMixin):
+    """File Group resource for working with user-uploaded group of files.
+
+    It can take group id or group CDN url::
+
+        >>> file_group = FileGroup('0513dda0-582f-447d-846f-096e5df9e2bb~2')
+
+    You can iterate ``file_group`` or get ``File`` instance by key::
+
+        >>> [file_ for file_ in file_group]
+        [<uploadcare.File 6c5e9526-b0fe-4739-8975-72e8d5ee6342>, None]
+        >>> file_group[0]
+        <uploadcare.File 6c5e9526-b0fe-4739-8975-72e8d5ee6342>
+        >>> len(file_group)
+        2
+
+    But slicing is not supported because ``FileGroup`` is immutable::
+
+        >>> file_group[:]
+        TypeError: slicing is not supported
+
+    If file was deleted then you will get ``None``::
+
+        >>> file_group[1]
+        None
+
+    """
+
+    def __init__(self, cdn_url_or_group_id):
+        matches = GROUP_ID_REGEX.search(cdn_url_or_group_id)
+
+        if not matches:
+            raise InvalidParamError("Couldn't find group id")
+
+        files_qty = int(matches.groupdict()["files_qty"])
+        if files_qty <= 0:
+            raise InvalidParamError("Couldn't find group id")
+
+        self.id = matches.groupdict()["group_id"]
+
+        self._files_qty = files_qty
+        self._info_cache = None
+
+    def __repr__(self):
+        return "<uploadcare.FileGroup {0}>".format(self.id)
+
+    def __str__(self):
+        return self.cdn_url
+
+    def __len__(self):
+        return self._files_qty
+
+    def __getitem__(self, key):
+        """Returns file from group by key as ``File`` instance."""
+        if isinstance(key, slice):
+            raise TypeError("slicing is not supported")
+        else:
+            file_info = self.info()["files"][key]
+            if file_info is not None:
+                return File.construct_from(file_info)
+
+    @property
+    def _api_uri(self):
+        return "groups/{0}/".format(self.id)
```

Comment:
here and below switch to f string for consistency

Reply:
done